harvester_e2e_tests/integrations/test_4_vm_backup_restore.py

```diff
@@ -626,6 +626,220 @@ def test_restore_replace_with_vm_shutdown_command(
         )
 
 
+@pytest.mark.p0
+@pytest.mark.backup_target
+@pytest.mark.parametrize(
+    "backup_config", [
+        pytest.param("S3", marks=pytest.mark.S3),
+        pytest.param("NFS", marks=pytest.mark.NFS)
+    ],
+    indirect=True)
+class TestBackupRestoreWithSnapshot:
+
+    @pytest.mark.dependency()
+    def test_connection(self, api_client, backup_config, config_backup_target):
+        code, data = api_client.settings.backup_target_test_connection()
+        assert 200 == code, f'Failed to test backup target connection: {data}'
+
+    @pytest.mark.dependency(depends=["TestBackupRestoreWithSnapshot::test_connection"], param=True)
+    def tests_backup_vm(self, api_client, wait_timeout, backup_config, base_vm_with_data):
```

Comment:
same as above, it looks you are going to reuse the test case?

Reply:
Thanks for the suggestion. 
The reason to have the test connection running again is to provide a new backup for all the test cases inside the new test class `TestBackupRestoreWithSnapshot`.

According to the test result in the follow up comment
https://github.com/harvester/tests/pull/1384#issuecomment-2275375643

And consider the trade off of pros and cons, may I keep these tests in the new test class. Many thanks.

Reply:
if you want to decouple new test cases without using existed one, you can just add another fixture to spawn up VM and do the backup (as it is the prerequisite of new test cases).

The problem here is, if you added those two duplicated (totally same as previous class) test cases, which means the total of test cases in report is not reflect to what we actually implemented and covered tests.

to add them after existed test cases would be more easy than create another fixture, that's why I suggested so.

Reply:
@TachunLin, do we have any ongoing work items here?
the current point is we should not have 2 same tests
* `TestBackupRestore:tests_backup_vm`
* `TestBackupRestoreWithSnapshot:tests_backup_vm`

I think one way is to make `TestBackupRestoreWithSnapshot:tests_backup_vm` a class fixture.

Reply:
@TachunLin ,
Per discussed, let you know that harvester/tests/issues/1462 has been fixed so there should be no connection fail.
You can continue surveying the refactor of this task item, thx.

Reply:
Thanks @albinsun for fixing the connection failed issue. 
According to the discussion and based on the [review comment](https://github.com/harvester/tests/pull/1384#discussion_r1709900479). 

I think it would be better moving the new snapshot restore test cases right after existed test cases in the same `TestBackupRestore`. It would solve the following concerns: 

1.  No reuse and duplication of `test_connection` and `tests_backup_vm`
2.  Save time without running the take backup action twice 

I already made the fix and tested working on the local to remote ECM lab machine
  ![image](https://github.com/user-attachments/assets/4bcec82e-8090-416c-b6d7-1e5badb70a24)

Will try again on the main Jenkins jobs to double confirm

Reply:
Tested `PASS` on the main Jenkins Harvester run test jobs [#102](http://172.19.99.74/job/harvester-runtests/102/)

All the new snapshot restore tests can execute well inside the `TestBackupRestore` test class. 
http://172.19.99.74/job/harvester-runtests/102/TestReport/

![image](https://github.com/user-attachments/assets/1306f443-c0fd-47f3-a29a-4db3081996f2)

The S3 related tests are blocked to proceed due to the environment broken caused by vpn disconnection yesterday.